services/command-router-quarkus/src/main/java/org/eclipse/hono/commandrouter/quarkus/Application.java

```diff
@@ -180,20 +183,19 @@ void setKafkaClientMetricsSupport(final KafkaMetricsOptions options) {
     }
 
     @Inject
-    void setKafkaClientOptions(final KafkaClientOptions options) {
-        this.kafkaProducerConfig = new MessagingKafkaProducerConfigProperties();
-        this.kafkaProducerConfig.setCommonClientConfig(options.commonClientConfig());
-        this.kafkaProducerConfig.setProducerConfig(options.producerConfig());
+    void setKafkaClientOptions(
+            @ConfigMapping(prefix = "hono.kafka") final CommonKafkaClientOptions commonOptions,
+            @ConfigMapping(prefix = "hono.kafka.commandOut") final KafkaProducerOptions producerOptions,
+            @ConfigMapping(prefix = "hono.kafka.commandIn") final KafkaConsumerOptions consumerOptions,
```

Comment:
In the Command Router, a Kafka producer is used for "command_internal" and "command_response" messages.
The Kafka consumer is used for "command" messages.
Therefore, I think it would be better, to use the corresponding names in the prefixes (like also in the adapters), i.e.:
```
@ConfigMapping(prefix = "hono.kafka.commandInternal") final KafkaProducerOptions commandInternalProducerOptions,
@ConfigMapping(prefix = "hono.kafka.commandResponse") final KafkaProducerOptions commandResponseProducerOptions,
@ConfigMapping(prefix = "hono.kafka.command") final KafkaConsumerOptions consumerOptions,
```

Reply:
Are you suggesting that I refactor the class `KafkaBasedCommandConsumerFactoryImpl` to take two separate producers in this PR?

Reply:
Ok, this PR is big enough already. This should be done in a subsequent PR then.